### PR TITLE
Fix bug on `TemporalModel.sample_time()`

### PR DESCRIPTION
**Pull request by fabiopintore**
This PR should fix a big in the `sample_time` function of `TemporalModel`, as reported in #4063 .

### CODE REVIEW

File: gammapy/modeling/models/tests/test_temporal.py
Comment (adonath) [inline gammapy/modeling/models/tests/test_temporal.py:78]: I think this could be split out as a separate test...

File: gammapy/modeling/models/tests/test_temporal.py
Comment (adonath) [inline gammapy/modeling/models/tests/test_temporal.py:85]: As the light curve is Gaussian here, maybe check both: the mean and standard deviation?

**Review by adonath — CHANGES_REQUESTED** [review]
Thanks @fabiopintore, this PR simplifies the implementation a lot! I have left two minor inline comments on the structure of the tests...

**Review by adonath — APPROVED** [review]
Thanks @fabiopintore! No further comments from my side. Before merging we need a second review...

**Review by registerrier — APPROVED** [review]
Thanks @fabiopintore . This looks indeed much better!

I have no specific comment.

**Comment by adonath** [comment]
Thanks @fabiopintore!

**Pull request merged**
